### PR TITLE
Make V Monoid

### PR DESCRIPTION
**Pull request by safareli**

### CODE REVIEW

File: src/Data/Validation/Semigroup.purs [hunk 1]
```diff
@@ -70,3 +71,9 @@ instance applyV :: (Semigroup err) => Apply (V err) where
 
 instance applicativeV :: (Semigroup err) => Applicative (V err) where
   pure = Valid
+
+instance semigroupV :: (Semigroup err, Semigroup a) => Semigroup (V err a) where
+  append x y = append <$> x <*> y
```
Comment (paf31) [inline src/Data/Validation/Semigroup.purs]: Can use `lift2` here.

File: src/Data/Validation/Semiring.purs [hunk 2]
```diff
@@ -77,6 +78,12 @@ instance applyV :: (Semiring err) => Apply (V err)  where
 instance applicativeV :: (Semiring err) => Applicative (V err) where
   pure = Valid
 
+instance semigroupV :: (Semiring err, Semigroup a) => Semigroup (V err a) where
+  append x y = append <$> x <*> y
```
Comment (paf31) [inline src/Data/Validation/Semiring.purs]: Same

File: src/Data/Validation/Semigroup.purs (hunk 1 above)
Comment (Profpatsch) [inline src/Data/Validation/Semigroup.purs]: Out of interest: is `lift2` faster?

Comment (safareli) [inline src/Data/Validation/Semigroup.purs]: I guess not, but difference should be really small right?
Also javascript runtime could optimise lift2 if it's called too often.

**Comment by safareli** [comment]
@paf31 @kRITZCREEK  ping

**Comment by paf31** [comment]
Thanks!

**Pull request merged**